src/core/jsdoc-tsd-parser.ts

```diff
@@ -127,7 +134,22 @@ export class JSDocTsdParser {
 
 					case "class":
 						// IClassDoclet with kind 'class'
-						parsedItem = this.parseClass(item as IClassDoclet);
+						if (this.resultItems[item.longname].length === 1) {
+
+							// class is already created, only add the constructor to the class
+							parsedItem = this.parseClass(item as IClassDoclet, this.resultItems[item.longname][0] as dom.ClassDeclaration);
+
+							// class is already added
+							addItem = false;
+						} else {
+
+							// create new class
+							parsedItem = this.parseClass(item as IClassDoclet);
+						}
+
+						// jsDocComment is already added to the constructors
+						// @classdesc is alrady added to the class
+						addJsDocComment = false;

```

Comment:
do we need this variable? If the class was already added, it won't be added again and the description will not be changed because you set the addItem to false?

Reply:
We need this variable in the case when the class is added. Because `item.comment` will be added to the constructor we don't need to add it to the class declaration. Additional, the class declaration already contains the descritpion from the `@classdesc` tag and we don't want to overwrite this special class description.

Reply:
okay, got it!